src/formatter-settings/index.ts

```diff
@@ -4,27 +4,347 @@
+            cloneElement.setAttribute("id", id);
+            cloneElement.setAttribute("value", value);
+            const newString: string = new XMLSerializer().serializeToString(cloneElement);
+            const edit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
+            edit.insert(document.uri, new vscode.Position(cloneElement.lineNumber - 1, cloneElement.columnNumber - 1 + originString.length), EOL + " ".repeat(cloneElement.columnNumber - 1) + newString);
+            vscode.workspace.applyEdit(edit);
+            cloneElement.lineNumber++;
+            this.profileElements.set(id, cloneElement);
+            this.profileSettingsArray.push(cloneElement);
+        } else {
+            // edit a current setting in the profile
+            const originSetting: string = new XMLSerializer().serializeToString(profileElement);
+            const line: number = profileElement.lineNumber;
+            const column: number = profileElement.columnNumber;
+            await profileElement.setAttribute("value", value);
+            const newSetting: string = new XMLSerializer().serializeToString(profileElement);
+            const edit: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
+            edit.replace(document.uri, new vscode.Range(new vscode.Position(line - 1, column - 1), new vscode.Position(line - 1, column - 1 + originSetting.length)), newSetting);
+            vscode.workspace.applyEdit(edit);
+        }
+    }
+
+    public async showFormatterSettingsEditor() {
+        if (!this.settingsUrl) {
+            vscode.window.showInformationMessage("No available Java Formatter Profile found, do you want to create a default profile?",
+                "Yes", "No").then((messageResult) => {
+                    if (messageResult === "Yes") {
+                        this.addFormatter();
+                    }
+                });
+        } else {
+            if (this.isRemote(this.settingsUrl)) {
+                await vscode.window.showInformationMessage("The active formatter profile is in remote URL, please change to a local file and try to reopen this setting again.");
+                return;
+            }
+            const profilePath = await this.getProfilePath(this.settingsUrl);
+            vscode.commands.executeCommand("vscode.openWith", vscode.Uri.file(profilePath), "java.formatterSettingsEditor");
+        }
+    }
+
+    private isRemote(path: string): boolean {
+        return path !== null && path.startsWith("http:/") || path.startsWith("https:/");
+    }
+
+    private async addFormatter(): Promise<void> {
+        const defaultProfile: string = path.join(this.context.extensionPath, "webview-resources", "formatter.xml");
+        let profilePath: string;
+        let relativePath: string | undefined;
+        const hasWorkspace: boolean = vscode.workspace.workspaceFolders !== undefined && vscode.workspace.workspaceFolders.length > 0;
```

Comment:
const workspaceFolder = vscode.workspace.workspaceFolders

And check the length of `workspaceFolder`